torch/_dynamo/variables/tensor.py

```diff
@@ -720,6 +722,55 @@ def call_method(
     ) -> "VariableTracker":
         unimplemented(f"numpy_ndarray.{name}()")
 
+    @staticmethod
+    def reconstruct_ndarray_before_return(codegen, output_graph) -> List[Instruction]:
+        """
+        Check if return value is torch_np.ndarray if so convert it to numpy.ndarray.
+        The equivalent Python code looks like this:
+        def f(x):
+            ___tmp_0 = __compiled_fn_0(x)
+            if isinstance(___tmp_0, torch_np.ndarray):
+                return ___tmp_0.tensor.numpy()
+            else:
+                return ___tmp_0
+        """
+        if not config.numpy_ndarray_as_tensor:
+            return []
+        from torch._dynamo.bytecode_transformation import create_instruction
+
+        # b.tensor.numpy()
+        var = output_graph.new_var()
+        from torch_np import ndarray
+
+        check = [
+            codegen.create_store(var),
+            codegen.create_load_global("isinstance", False, add=True),
```

Comment:
The user could theoretically shadow this name in their global scope.

Reply:
Yea, you can use this helper function to mitigate:
https://github.com/pytorch/pytorch/blob/ee5f09ab802a04cc966829230c236e9132041ab7/torch/_dynamo/codegen.py#L291

but a better way is to replace these bytecode by ```to_numpy_helper```.